src/System.Buffers.Primitives/System/Buffers/OwnedBuffer.cs

```diff
@@ -11,118 +11,43 @@ namespace System.Buffers
-
-        protected OwnedBuffer(T[] array, int arrayOffset, int length, IntPtr pointer = default(IntPtr))
-        {
-            _id = InitializedId;
-            Initialize(array, arrayOffset, length, pointer);
-        }
+        public abstract Span<T> GetSpan(int index, int length);
 
         public Buffer<T> Buffer => new Buffer<T>(this, 0, Length);
-        public ReadOnlyBuffer<T> ReadOnlyBuffer => new ReadOnlyBuffer<T>(this, 0, Length);
 
-        public Span<T> Span
-        {
-            [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            get {
-                var array = Array;
-                if (array != null)
-                    return new Span<T>(array, _arrayIndex, _length);
-                else unsafe {
-                    return new Span<T>(Pointer.ToPointer(), Length);
-                }
-            }
-        }
+        public ReadOnlyBuffer<T> ReadOnlyBuffer => new ReadOnlyBuffer<T>(this, 0, Length);
 
         public static implicit operator OwnedBuffer<T>(T[] array) => new Internal.OwnedArray<T>(array);
 
-        protected bool TryGetArrayCore(out ArraySegment<T> buffer)
-        {
-            if (Array == null) {
-                buffer = default(ArraySegment<T>);
-                return false;
-            }
-
-            buffer = new ArraySegment<T>(Array, Offset, Length);
-            return true;
-        }
-
-        protected unsafe bool TryGetPointerCore(out void* pointer)
-        {
-            if (Pointer == IntPtr.Zero) {
-                pointer = null;
-                return false;
-            }
-
-            pointer = Pointer.ToPointer();
-            return true;
-        }
-
         #region Lifetime Management
-        protected void Initialize(T[] array, int arrayOffset, int length, IntPtr pointer = default(IntPtr))
-        {
-            Contract.Requires(array != null || pointer != IntPtr.Zero);
-            Contract.Requires(array == null || arrayOffset + length <= array.Length);
-            if (!IsDisposed && Id!=InitializedId) {
-                throw new InvalidOperationException("this instance has to be disposed to initialize");
-            }
-
-            _id = (int)Interlocked.Increment(ref _nextId);
-            _array = array;
-            _arrayIndex = arrayOffset;
-            _length = length;
-            _pointer = pointer;
-            _referenceCount = 0;
-        }
+        public bool IsDisposed => _disposed;
 
         public void Dispose()
         {
-            Interlocked.Exchange(ref _id,  FreedId);
             if (HasOutstandingReferences) throw new InvalidOperationException("outstanding references detected.");
             Dispose(true);
         }
 
         protected virtual void Dispose(bool disposing)
-        { 
-            _id = FreedId;
-            _array = null;
-            _pointer = IntPtr.Zero;
-            _length = 0;
-            _arrayIndex = 0;
+        {
+            _disposed = disposing;
```

Comment:
Should the dispose method be abstract with no default implementation? What is the right approach?